### PR TITLE
Drop notebook 6

### LINKED ISSUE #4779: Drop support for jupyter notebook < 7

### PR DESCRIPTION
**Pull request by marthacryan**
**Note: this PR is on top of the changes from #4823 because the plotly.js version used in plotly.py is based on the jupyterlab extension code, and we needed a new place to store the plotly.js version. The widget seemed like a good place because all of the other javascript is removed in this PR.**

Fixes #4779. 

It appears that the entire `packages/javascript/jupyterlab_plotly` directory contains code that is either used to support very old versions of jupyter lab/notebook/voila, or is not used at all. This PR removes the entire directory in favor of having a very small `packages/python/plotly/js` directory that contains only the widget code (see #4663 for adding the widget code into that directory).

The last time that the jupyter extension was significantly changed was 3 years ago in #3142. @fcollonval described the files in the `packages/javascript/jupyterlab_plotly` directory in [this comment](https://github.com/plotly/plotly.py/pull/3142#issuecomment-819637791). He did add support for federated extensions (new jupyter lab/notebook build), but I wasn't able to find evidence that having an extension installed in jupyterlab >=3 or jupyter notebook >= 7 is even necessary for plotly. @fcollonval if you know of any reason that that isn't true, please let me know! 

To test:
* Checkout branch
* Create a new python env (conda, mamba, pipenv, etc)
* `git clean -f -x -d`
* `cd packages/python/plotly`
* `pip install -e .`
* `pip install jupyter`
* Try it out in jupyter lab: `jupyter lab`
* Try it out in jupyter notebook: `jupyter notebook`

### CODE REVIEW

**Comment by fcollonval** [comment]
> I wasn't able to find evidence that having an extension installed in jupyterlab >=3 or jupyter notebook >= 7 is even necessary for plotly

This is correct because the produced output is using html that will includes plotly.js the first time:

https://github.com/plotly/plotly.py/blob/7bcbb20c4fe7ec24461d4c16a4d8df14fdcce73b/packages/python/plotly/plotly/io/_base_renderers.py#L368

Dropping the [mimerenderer factory](https://github.com/plotly/plotly.py/blob/7bcbb20c4fe7ec24461d4c16a4d8df14fdcce73b/packages/javascript/jupyterlab-plotly/src/plotly-renderer.ts) will prevent `plotly.json` file to be opened in JupyterLab.

**Comment by marthacryan** [comment]
@fcollonval Thank you for the response! Also hi!

**Comment by archmoj** [comment]
This is looking good to me.
@marthacryan Would you please add a changelog entry?

**Comment by LiamConnors** [comment]
@marthacryan, testing the `FigureWidget` in Notebook 7 and I get a JavaScript error:

Code is

```
import plotly.graph_objects as go

f = go.FigureWidget()
f

```

image

I have `anywidget=0.9.13`

**Comment by marthacryan** [comment]
@LiamConnors Looks like this wasn't happening for me because I hadn't added the `npm install && npm run build` step into `pip install -e .`, so the necessary javascript wasn't generated! I'll look into adding that

**Comment by LiamConnors** [comment]
In VS Code, I see this:

image

Not really sure what to make of it. I'm using the same environment as in a notebook launched in the browser.

File: README.md [hunk 1]
```diff
@@ -111,26 +111,21 @@ conda install "jupyterlab>=3" "ipywidgets>=7.6"
...
-jupyter labextension install jupyterlab-plotly@5.24.1 @jupyter-widgets/jupyterlab-manager
-```
-
 Please check out our [Troubleshooting guide](https://plotly.com/python/troubleshooting/) if you run into any problems with JupyterLab.
 
-### Jupyter Notebook Support
+### Jupyter Widget Support
 
-For use in the Jupyter Notebook, install the `notebook` and `ipywidgets`
+For use as a Jupyter widget, install `jupyter` and `anywidget`
```
Comment (LiamConnors) [inline README.md:118]: The section above this talks about "JupyterLab Support" and should now mention anywidget. Wondering if this section should stay as "Jupyter Notebook Support"

Comment (marthacryan) [reply README.md:118]: I think maybe they should be combined. The usage is almost identical for jupyter Notebook and JupyterLab

File: packages/python/plotly/plotly/io/_renderers.py [hunk 2]
```diff
@@ -395,6 +398,19 @@ def show(fig, renderer=None, validate=True, **kwargs):
...
+
+        if "jupyter-notebook" in parent_process and jupyter_notebook.__version__ < "7":
+            # Add warning about upgrading notebook
+            warnings.warn(
+                f"Plotly version >= 6 requires Jupyter Notebook >= 7 but you have {jupyter_notebook.__version__} installed.\n To upgrade Jupyter Notebook, please run `pip install notebook --upgrade`."
+            )
+        elif "jupyter-lab" in parent_process and jupyter_lab.__version__ < "3":
+            # Add warning about upgrading jupyterlab
+            warnings.warn(
+                f"Plotly version >= 6 requires JupyterLab >= 3 but you have {jupyter_lab.__version__} installed. To upgrade JupyterLab, please run `pip install jupyterlab --upgrade`."
```
Comment (marthacryan) [inline packages/python/plotly/plotly/io/_renderers.py:411]: Maybe put this in `FigureWidget` instead?

**Pull request merged**